_posts/2022-11-16-#25261.md

```diff
@@ -7,17 +7,82 @@ authors: [natanleung]
 components: ["rpc"]
 host: larryruane
 status: upcoming
-commit:
+commit: 053ccf0468e477283e80f78cc095ffb83bff9b95
 ---
 
 _Notes and questions to follow soon!_
 
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+1. The [`getblockheader` RPC](https://developer.bitcoin.org/reference/rpc/getblockheader.html)
+returns a given block header, given its hash. The header data is returned as
+a JSON object (verbose arg `true`, which is default) or in raw hex form (verbose `false`).
+
+1. The [REST interface](https://github.com/bitcoin/bitcoin/blob/master/doc/REST-interface.md)
+provides another way to query a `bitcoind` node. It's not enabled by default; specify the
+`bitcoind -rest` command-line option or `rest=1` in the config file to enable this service.
+
+1. REST also provides a
+[blockheader](https://github.com/bitcoin/bitcoin/blob/master/doc/REST-interface.md#blockheaders)
+method to fetch block headers; on mainnet, try:
+```
+curl -s localhost:8332/rest/headers/00000000000000000006c042058f7ff60003ae9a96ca2ac3065d91221b00f547.json
+```
+This returns five block headers beginning with the specified block hash. You can specify
+the number of headers by appending `?count=nnn` to the URL.
+The maximum number of results is 2000.
+
+1. This PR proposes to allow the `getblockheader` RPC to return more than one header by adding an
+optional `count` argument.
+
 ## Questions
-1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
--->
+1. Did you review the PR?
+[Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+
+1. Note that the `count` argument follows the `verbose` argument. (Why?)
+Suppose you do not want to specify the `verbose` argument (that is, you prefer the default),
+but you do want to specify `count`. Is there a way to do that?
+
+1. Why is the type of the `count` argument an
+[`RPCArg::Type::AMOUNT`](https://github.com/bitcoin/bitcoin/pull/25261/files#diff-decae4be02fb8a47ab4557fe74a9cb853bdfa3ec0fa1b515c0a1e5de91f4ad0bR506)
+rather than a `RPCArg::Type::NUM` as would seem more natural?
+
+1. The default number of headers to return is 1, yet there is a difference
+between specifying `count=1` and not specifying a count. What is this difference?
+Why do these behave differently, and should they?
+
+1. Why is the count limited to 2000? Do you agree with this limit?
+What are the tradeoffs?
+
+1. What happens if there are fewer than `count` headers beginning with the
+one specified?
```

Comment:
It's not preceding, it's following. For example, if you specify the block hash of the tip and ask for 2 headers, it can only give you one. So it gives you one. But now that I think about it, it's pretty obvious, and we have arguably too many questions, so I'm just going to drop this question, it's both obvious yet also a bit too detailed (corner case).